{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "c6d3616e-8579-4225-bb29-79aebf9496ff",
   "metadata": {},
   "source": [
    "### Шаг 0: скачивание данных"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f8eef7bb-daaf-472e-9cab-e10b0321d90a",
   "metadata": {},
   "source": [
    "В отдельной папке скачать все архивы и распковал их командой \n",
    "\n",
    "gunzip *.gz\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "007ff72d-8080-425e-8c6e-0cfacc87c25a",
   "metadata": {},
   "source": [
    "### Шаг 1: Контроль качества ридов (15 баллов)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "27a145e9-9f0a-4b50-931f-b82444785f18",
   "metadata": {},
   "source": [
    "Создаём окружение\n",
    "\n",
    "conda create -n DD_fastqc_h-c bioconda fastqc\n",
    "\n",
    "___________________________________________________________________________\n",
    "\n",
    "\n",
    "Downloading and Extracting Packages:\n",
    "\n",
    "openjdk-22.0.1 \n",
    "\n",
    "fastqc-0.12.1   \n",
    "\n",
    "ibexpat-2.6.4    \n",
    "\n",
    "___________________________________________________________________________\n",
    "FastQC v0.12.1\n",
    "\n",
    "conda activate DD_fastqc_h\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "038a7204-2bb8-4c06-95d7-079cfff1ae6c",
   "metadata": {},
   "source": [
    "Запускаем fastqc\n",
    "\n",
    "(DD_fastqc_h) mhprs@Seraph:~/PISH/Data_driven/DD_2024_private/hw2/data$ fastqc SRR292862_S2_L001_R1_001.fastq SRR292862_S2_L001_R2_001.fastq SRR292678sub_S1_L001_R1_001.fastq SRR292678sub_S1_L001_R2_001.fastq SRR292770_S1_L001_R1_001.fastq SRR292770_S1_L001_R2_001.fastq -o ../fastqc/\n",
    "\n",
    "\n",
    "null\n",
    "null\n",
    "Started analysis of SRR292862_S2_L001_R1_001.fastq\n",
    "null\n",
    "null\n",
    "null\n",
    "null\n",
    "Approx 5% complete for SRR292862_S2_L001_R1_001.fastq\n",
    "Approx 10% complete for SRR292862_S2_L001_R1_001.fastq\n",
    "Approx 15% complete for SRR292862_S2_L001_R1_001.fastq\n",
    "Approx 20% complete for SRR292862_S2_L001_R1_001.fastq\n",
    "Approx 25% complete for SRR292862_S2_L001_R1_001.fastq\n",
    "Approx 30% complete for SRR292862_S2_L001_R1_001.fastq\n",
    "Approx 35% complete for SRR292862_S2_L001_R1_001.fastq\n",
    "Approx 40% complete for SRR292862_S2_L001_R1_001.fastq\n",
    "Approx 45% complete for SRR292862_S2_L001_R1_001.fastq\n",
    "Approx 50% complete for SRR292862_S2_L001_R1_001.fastq\n",
    "Approx 55% complete for SRR292862_S2_L001_R1_001.fastq\n",
    "Approx 60% complete for SRR292862_S2_L001_R1_001.fastq\n",
    "Approx 65% complete for SRR292862_S2_L001_R1_001.fastq\n",
    "Approx 70% complete for SRR292862_S2_L001_R1_001.fastq\n",
    "Approx 75% complete for SRR292862_S2_L001_R1_001.fastq\n",
    "Approx 80% complete for SRR292862_S2_L001_R1_001.fastq\n",
    "Approx 85% complete for SRR292862_S2_L001_R1_001.fastq\n",
    "Approx 90% complete for SRR292862_S2_L001_R1_001.fastq\n",
    "Approx 95% complete for SRR292862_S2_L001_R1_001.fastq\n",
    "Analysis complete for SRR292862_S2_L001_R1_001.fastq\n",
    "Started analysis of SRR292862_S2_L001_R2_001.fastq\n",
    "Approx 5% complete for SRR292862_S2_L001_R2_001.fastq\n",
    "Approx 10% complete for SRR292862_S2_L001_R2_001.fastq\n",
    "Approx 15% complete for SRR292862_S2_L001_R2_001.fastq\n",
    "Approx 20% complete for SRR292862_S2_L001_R2_001.fastq\n",
    "Approx 25% complete for SRR292862_S2_L001_R2_001.fastq\n",
    "Approx 30% complete for SRR292862_S2_L001_R2_001.fastq\n",
    "Approx 35% complete for SRR292862_S2_L001_R2_001.fastq\n",
    "Approx 40% complete for SRR292862_S2_L001_R2_001.fastq\n",
    "Approx 45% complete for SRR292862_S2_L001_R2_001.fastq\n",
    "Approx 50% complete for SRR292862_S2_L001_R2_001.fastq\n",
    "Approx 55% complete for SRR292862_S2_L001_R2_001.fastq\n",
    "Approx 60% complete for SRR292862_S2_L001_R2_001.fastq\n",
    "Approx 65% complete for SRR292862_S2_L001_R2_001.fastq\n",
    "Approx 70% complete for SRR292862_S2_L001_R2_001.fastq\n",
    "Approx 75% complete for SRR292862_S2_L001_R2_001.fastq\n",
    "Approx 80% complete for SRR292862_S2_L001_R2_001.fastq\n",
    "Approx 85% complete for SRR292862_S2_L001_R2_001.fastq\n",
    "Approx 90% complete for SRR292862_S2_L001_R2_001.fastq\n",
    "Approx 95% complete for SRR292862_S2_L001_R2_001.fastq\n",
    "Analysis complete for SRR292862_S2_L001_R2_001.fastq\n",
    "Started analysis of SRR292678sub_S1_L001_R1_001.fastq\n",
    "Approx 5% complete for SRR292678sub_S1_L001_R1_001.fastq\n",
    "Approx 10% complete for SRR292678sub_S1_L001_R1_001.fastq\n",
    "Approx 15% complete for SRR292678sub_S1_L001_R1_001.fastq\n",
    "Approx 20% complete for SRR292678sub_S1_L001_R1_001.fastq\n",
    "Approx 25% complete for SRR292678sub_S1_L001_R1_001.fastq\n",
    "Approx 30% complete for SRR292678sub_S1_L001_R1_001.fastq\n",
    "Approx 35% complete for SRR292678sub_S1_L001_R1_001.fastq\n",
    "Approx 40% complete for SRR292678sub_S1_L001_R1_001.fastq\n",
    "Approx 45% complete for SRR292678sub_S1_L001_R1_001.fastq\n",
    "Approx 50% complete for SRR292678sub_S1_L001_R1_001.fastq\n",
    "Approx 55% complete for SRR292678sub_S1_L001_R1_001.fastq\n",
    "Approx 60% complete for SRR292678sub_S1_L001_R1_001.fastq\n",
    "Approx 65% complete for SRR292678sub_S1_L001_R1_001.fastq\n",
    "Approx 70% complete for SRR292678sub_S1_L001_R1_001.fastq\n",
    "Approx 75% complete for SRR292678sub_S1_L001_R1_001.fastq\n",
    "Approx 80% complete for SRR292678sub_S1_L001_R1_001.fastq\n",
    "Approx 85% complete for SRR292678sub_S1_L001_R1_001.fastq\n",
    "Approx 90% complete for SRR292678sub_S1_L001_R1_001.fastq\n",
    "Approx 95% complete for SRR292678sub_S1_L001_R1_001.fastq\n",
    "Analysis complete for SRR292678sub_S1_L001_R1_001.fastq\n",
    "Started analysis of SRR292678sub_S1_L001_R2_001.fastq\n",
    "Approx 5% complete for SRR292678sub_S1_L001_R2_001.fastq\n",
    "Approx 10% complete for SRR292678sub_S1_L001_R2_001.fastq\n",
    "Approx 15% complete for SRR292678sub_S1_L001_R2_001.fastq\n",
    "Approx 20% complete for SRR292678sub_S1_L001_R2_001.fastq\n",
    "Approx 25% complete for SRR292678sub_S1_L001_R2_001.fastq\n",
    "Approx 30% complete for SRR292678sub_S1_L001_R2_001.fastq\n",
    "Approx 35% complete for SRR292678sub_S1_L001_R2_001.fastq\n",
    "Approx 40% complete for SRR292678sub_S1_L001_R2_001.fastq\n",
    "Approx 45% complete for SRR292678sub_S1_L001_R2_001.fastq\n",
    "Approx 50% complete for SRR292678sub_S1_L001_R2_001.fastq\n",
    "Approx 55% complete for SRR292678sub_S1_L001_R2_001.fastq\n",
    "Approx 60% complete for SRR292678sub_S1_L001_R2_001.fastq\n",
    "Approx 65% complete for SRR292678sub_S1_L001_R2_001.fastq\n",
    "Approx 70% complete for SRR292678sub_S1_L001_R2_001.fastq\n",
    "Approx 75% complete for SRR292678sub_S1_L001_R2_001.fastq\n",
    "Approx 80% complete for SRR292678sub_S1_L001_R2_001.fastq\n",
    "Approx 85% complete for SRR292678sub_S1_L001_R2_001.fastq\n",
    "Approx 90% complete for SRR292678sub_S1_L001_R2_001.fastq\n",
    "Approx 95% complete for SRR292678sub_S1_L001_R2_001.fastq\n",
    "Analysis complete for SRR292678sub_S1_L001_R2_001.fastq\n",
    "Started analysis of SRR292770_S1_L001_R1_001.fastq\n",
    "Approx 5% complete for SRR292770_S1_L001_R1_001.fastq\n",
    "Approx 10% complete for SRR292770_S1_L001_R1_001.fastq\n",
    "Approx 15% complete for SRR292770_S1_L001_R1_001.fastq\n",
    "Approx 20% complete for SRR292770_S1_L001_R1_001.fastq\n",
    "Approx 25% complete for SRR292770_S1_L001_R1_001.fastq\n",
    "Approx 30% complete for SRR292770_S1_L001_R1_001.fastq\n",
    "Approx 35% complete for SRR292770_S1_L001_R1_001.fastq\n",
    "Approx 40% complete for SRR292770_S1_L001_R1_001.fastq\n",
    "Approx 45% complete for SRR292770_S1_L001_R1_001.fastq\n",
    "Approx 50% complete for SRR292770_S1_L001_R1_001.fastq\n",
    "Approx 55% complete for SRR292770_S1_L001_R1_001.fastq\n",
    "Approx 60% complete for SRR292770_S1_L001_R1_001.fastq\n",
    "Approx 65% complete for SRR292770_S1_L001_R1_001.fastq\n",
    "Approx 70% complete for SRR292770_S1_L001_R1_001.fastq\n",
    "Approx 75% complete for SRR292770_S1_L001_R1_001.fastq\n",
    "Approx 80% complete for SRR292770_S1_L001_R1_001.fastq\n",
    "Approx 85% complete for SRR292770_S1_L001_R1_001.fastq\n",
    "Approx 90% complete for SRR292770_S1_L001_R1_001.fastq\n",
    "Approx 95% complete for SRR292770_S1_L001_R1_001.fastq\n",
    "Analysis complete for SRR292770_S1_L001_R1_001.fastq\n",
    "Started analysis of SRR292770_S1_L001_R2_001.fastq\n",
    "Approx 5% complete for SRR292770_S1_L001_R2_001.fastq\n",
    "Approx 10% complete for SRR292770_S1_L001_R2_001.fastq\n",
    "Approx 15% complete for SRR292770_S1_L001_R2_001.fastq\n",
    "Approx 20% complete for SRR292770_S1_L001_R2_001.fastq\n",
    "Approx 25% complete for SRR292770_S1_L001_R2_001.fastq\n",
    "Approx 30% complete for SRR292770_S1_L001_R2_001.fastq\n",
    "Approx 35% complete for SRR292770_S1_L001_R2_001.fastq\n",
    "Approx 40% complete for SRR292770_S1_L001_R2_001.fastq\n",
    "Approx 45% complete for SRR292770_S1_L001_R2_001.fastq\n",
    "Approx 50% complete for SRR292770_S1_L001_R2_001.fastq\n",
    "Approx 55% complete for SRR292770_S1_L001_R2_001.fastq\n",
    "Approx 60% complete for SRR292770_S1_L001_R2_001.fastq\n",
    "Approx 65% complete for SRR292770_S1_L001_R2_001.fastq\n",
    "Approx 70% complete for SRR292770_S1_L001_R2_001.fastq\n",
    "Approx 75% complete for SRR292770_S1_L001_R2_001.fastq\n",
    "Approx 80% complete for SRR292770_S1_L001_R2_001.fastq\n",
    "Approx 85% complete for SRR292770_S1_L001_R2_001.fastq\n",
    "Approx 90% complete for SRR292770_S1_L001_R2_001.fastq\n",
    "Approx 95% complete for SRR292770_S1_L001_R2_001.fastq\n",
    "Analysis complete for SRR292770_S1_L001_R2_001.fastq\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d20c4b72-b402-4881-82a8-de97a4f195b7",
   "metadata": {},
   "outputs": [],
   "source": [
    "Делаем коммит\n",
    "\n",
    "\n",
    "git add -A\n",
    "git status\n",
    "\n",
    "git commit -"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bbf7ad9d-22a5-40b0-9dc5-fbe7489cf371",
   "metadata": {},
   "source": [
    "### Шаг 2: Сборка генома (20 баллов)\n",
    "Создайте conda-окружение spades\n",
    "Установите SPAdes\n",
    "Выполните сборку генома используя только paired-end риды (SRR292678)\n",
    "Сохраните файлы contigs.fasta и scaffolds.fasta в папку hw2/spades\n",
    "Зафиксируйте команды в lab_journal.ipynb\n",
    "Сделайте коммит всех изменений\n",
    "Проверка:\n",
    "- Наличие непустых contigs.fasta и scaffolds.fasta в hw2/spades\n",
    "- Корректные записи в lab_journal.ipynb\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1ce07fe5-f4ea-48a4-becf-adfc84ee4a1b",
   "metadata": {},
   "source": [
    "Создаём окружение \n",
    "\n",
    "conda create -n DD_spades_h -c bioconda -c conda-forge spades"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "392b10b6-d1d3-4ec4-be23-2d5a1ee30ad2",
   "metadata": {},
   "source": [
    "$ spades.py --version\n",
    "\n",
    "/home/mhprs/miniconda3/envs/DD_spades_h/share/spades/spades_pipeline/support.py:488: SyntaxWarning: invalid escape sequence '\\d'\n",
    "\n",
    "  return [atoi(c) for c in re.split(\"(\\d+)\", text)]\n",
    "  \n",
    "SPAdes genome assembler v4.0.0"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "49ef6bf9-8351-434a-ac6d-675ea1eb8366",
   "metadata": {},
   "source": [
    "spades.py \\\n",
    "  --pe1-1 SRR292678sub_S1_L001_R1_001.fastq --pe1-2 SRR292678sub_S1_L001_R2_001.fastq \\\n",
    "  --pe2-1 SRR292770_S1_L001_R1_001.fastq --pe2-2 SRR292770_S1_L001_R2_001.fastq \\\n",
    "  --pe3-1 SRR292862_S2_L001_R1_001.fastq --pe3-2 SRR292862_S2_L001_R2_001.fastq \\\n",
    "  -o hw2/spades\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a519e0e0-3133-4c45-8f5a-fa7e43b3a7a0",
   "metadata": {},
   "source": [
    "вышла ошибка"
   ]
  },
  {
   "cell_type": "raw",
   "id": "f0bde9ca-8ff5-4c97-a897-0973e1d594bc",
   "metadata": {},
   "source": [
    "== Error ==  system call for: \"['/home/mhprs/miniconda3/envs/DD_spades_h/bin/spades-hammer', '/home/mhprs/PISH/Data_driven/DD_2024_private/hw2/data/hw2/spades/corrected/configs/config.info']\" finished abnormally, OS return value: -6\n",
    "None\n",
    "\n",
    "In case you have troubles running SPAdes, you can report an issue on our GitHub repository github.com/ablab/spades\n",
    "Please provide us with params.txt and spades.log files from the output directory.\n",
    "\n",
    "SPAdes log can be found here: /home/mhprs/PISH/Data_driven/DD_2024_private/hw2/data/hw2/spades/spades.log\n",
    "\n",
    "Thank you for using SPAdes! If you use it in your research, please cite:\n",
    "\n",
    "  Prjibelski, A., Antipov, D., Meleshko, D., Lapidus, A. and Korobeynikov, A., 2020. Using SPAdes de novo assembler. Current protocols in bioinformatics, 70(1), p.e102.\n",
    "  doi.org/10.1002/cpbi.102"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "30eb1958-181d-4224-a5e8-ea0e24a4d0fa",
   "metadata": {},
   "source": [
    "удалим временные файлы и заново запустим для одного(возможно проблема от того, что все оутпуты в одну папку)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "14784cc7-47bc-468a-85e3-4e251f033b06",
   "metadata": {},
   "source": [
    "======= SPAdes pipeline finished WITH WARNINGS!\n",
    "\n",
    "хоты бы finished\n",
    "\n",
    "\n",
    "commit\n",
    "\n",
    "\n",
    "Total 68 (delta 20), reused 0 (delta 0), pack-reused 0\n",
    "remote: Resolving deltas: 100% (20/20), completed with 2 local objects.\n",
    "remote: error: Trace: 2330b19ef0a3921a81518161d46395298e0c56dd7d5b30082c5a7bfb669b06df\n",
    "remote: error: See https://gh.io/lfs for more information.\n",
    "remote: error: File hw2/spades/corrected/SRR292678sub_S1_L001_R1_00100.0_0.cor.fastq.gz is 427.00 MB; this exceeds GitHub's file size limit of 100.00 MB\n",
    "remote: error: File hw2/spades/corrected/SRR292678sub_S1_L001_R2_00100.0_0.cor.fastq.gz is 427.57 MB; this exceeds GitHub's file size limit of 100.00 MB\n",
    "remote: error: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.\n",
    "To github.com:MPHRS/DD_2024_private.git\n",
    " ! [remote rejected] main -> main (pre-receive hook declined)\n",
    "error: failed to push some refs to 'github.com:MPHRS/DD_2024_private.git'\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "Решение с использованием Git LFS:\n",
    "\n",
    "    Установите Git LFS\n",
    "\n",
    "    Для начала, вам нужно установить Git LFS на вашу машину. Если он еще не установлен, выполните следующую команду:\n",
    "\n",
    "    Для Linux (Ubuntu/Debian):\n",
    "\n",
    "sudo apt-get install git-lfs\n",
    "\n",
    "Для macOS:\n",
    "\n",
    "brew install git-lfs\n",
    "\n",
    "Для Windows: Следуйте инструкциям на официальной странице Git LFS.\n",
    "\n",
    "Инициализация Git LFS в вашем репозитории\n",
    "\n",
    "После установки Git LFS, вам нужно инициализировать его в вашем репозитории:\n",
    "\n",
    "git lfs install\n",
    "\n",
    "Отслеживание больших файлов с помощью Git LFS\n",
    "\n",
    "Чтобы Git LFS отслеживал файлы, которые превышают лимит в 100 MB (например, ваши .fastq.gz файлы), выполните команду:\n",
    "\n",
    "git lfs track \"*.fastq.gz\"\n",
    "\n",
    "Это скажет Git LFS отслеживать все файлы с расширением .fastq.gz в вашем репозитории.\n",
    "\n",
    "Добавьте изменения в индекс Git и сделайте коммит\n",
    "\n",
    "После того как вы настроили отслеживание больших файлов, добавьте изменения в индекс и сделайте новый коммит:\n",
    "\n",
    "git add .gitattributes\n",
    "git add hw2/spades/corrected/SRR292678sub_S1_L001_R1_00100.0_0.cor.fastq.gz\n",
    "git add hw2/spades/corrected/SRR292678sub_S1_L001_R2_00100.0_0.cor.fastq.gz\n",
    "git commit -m \"Добавлены большие файлы с использованием Git LFS\"\n",
    "\n",
    "Пуш изменений в репозиторий\n",
    "\n",
    "Теперь, когда файлы отслеживаются с помощью Git LFS, вы можете снова попробовать выполнить пуш в репозиторий:\n",
    "\n",
    "git push\n",
    "\n",
    "(DD_spades_h) mhprs@Seraph:~/PISH/Data_driven/DD_2024_private$ git push\n",
    "Uploading LFS objects: 100% (2/2), 896 MB | 18 MB/s, done.\n",
    "Enumerating objects: 79, done.\n",
    "Counting objects: 100% (79/79), done.\n",
    "Delta compression using up to 8 threads\n",
    "Compressing objects: 100% (68/68), done.\n",
    "Writing objects: 100% (76/76), 854.69 MiB | 3.71 MiB/s, done.\n",
    "Total 76 (delta 24), reused 0 (delta 0), pack-reused 0\n",
    "remote: Resolving deltas: 100% (24/24), completed with 2 local objects.\n",
    "remote: error: Trace: fcd5e90b3b680a841a925e16eb14c03e36128f5d1340647c52a3816f814d049d\n",
    "remote: error: See https://gh.io/lfs for more information.\n",
    "remote: error: File hw2/spades/corrected/SRR292678sub_S1_L001_R1_00100.0_0.cor.fastq.gz is 427.00 MB; this exceeds GitHub's file size limit of 100.00 MB\n",
    "remote: error: File hw2/spades/corrected/SRR292678sub_S1_L001_R2_00100.0_0.cor.fastq.gz is 427.57 MB; this exceeds GitHub's file size limit of 100.00 MB\n",
    "remote: error: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.\n",
    "To github.com:MPHRS/DD_2024_private.git\n",
    " ! [remote rejected] main -> main (pre-receive hook declined)\n",
    "error: failed to push some refs to 'github.com:MPHRS/DD_2024_private.git'\n",
    "(DD_spades_h) mhprs@Seraph:~/PISH/Data_driven/DD_2024_private$ git status\n",
    "On branch main\n",
    "Your branch is ahead of 'origin/main' by 2 commits.\n",
    "  (use \"git push\" to publish your local commits)\n",
    "\n",
    "Changes not staged for commit:\n",
    "  (use \"git add <file>...\" to update what will be committed)\n",
    "  (use \"git restore <file>...\" to discard changes in working directory)\n",
    "        modified:   hw2/lab_journal.ipynb\n",
    "        modified:   hw2/spades/corrected/SRR292678sub_S1_L001_R_unpaired00.0_0.cor.fastq.gz\n",
    "\n",
    "no changes added to commit (use \"git add\" and/or \"git commit -a\")\n",
    "(DD_spades_h) mhprs@Seraph:~/PISH/Data_driven/DD_2024_private$ git reset --soft HEAD~2\n",
    "(DD_spades_h) mhprs@Seraph:~/PISH/Data_driven/DD_2024_private$ git status\n",
    "On branch main\n",
    "Your branch is up to date with 'origin/main'.\n",
    "\n",
    "Changes to be committed:\n",
    "  (use \"git restore --staged <file>...\" to unstage)\n",
    "        new file:   .gitattributes\n",
    "        modified:   hw2/lab_journal.ipynb\n",
    "        new file:   hw2/spades/K21/configs/careful_mda_mode.info\n",
    "        new file:   hw2/spades/K21/configs/careful_mode.info\n",
    "        new file:   hw2/spades/K21/configs/config.info\n",
    "        new file:   hw2/spades/K21/configs/construction.info\n",
    "        new file:   hw2/spades/K21/configs/detail_info_printer.info\n",
    "        new file:   hw2/spades/K21/configs/distance_estimation.info\n",
    "        new file:   hw2/spades/K21/configs/hmm_mode.info\n",
    "        new file:   hw2/spades/K21/configs/isolate_mode.info\n",
    "        new file:   hw2/spades/K21/configs/large_genome_mode.info\n",
    "        new file:   hw2/spades/K21/configs/mda_mode.info\n",
    "        new file:   hw2/spades/K21/configs/meta_mode.info\n",
    "        new file:   hw2/spades/K21/configs/metaplasmid_mode.info\n",
    "        new file:   hw2/spades/K21/configs/metaviral_mode.info\n",
    "        new file:   hw2/spades/K21/configs/pe_params.info\n",
    "        new file:   hw2/spades/K21/configs/plasmid_mode.info\n",
    "        new file:   hw2/spades/K21/configs/rna_mode.info\n",
    "        new file:   hw2/spades/K21/configs/rnaviral_mode.info\n",
    "        new file:   hw2/spades/K21/configs/sewage_mode.info\n",
    "        new file:   hw2/spades/K21/configs/simplification.info\n",
    "        new file:   hw2/spades/K21/configs/toy.info\n",
    "        new file:   hw2/spades/K21/final.lib_data\n",
    "        new file:   hw2/spades/K21/simplified_contigs/contigs.off\n",
    "        new file:   hw2/spades/K21/simplified_contigs/contigs.seq\n",
    "        new file:   hw2/spades/K21/simplified_contigs/contigs_info\n",
    "        new file:   hw2/spades/K33/configs/careful_mda_mode.info\n",
    "        new file:   hw2/spades/K33/configs/careful_mode.info\n",
    "        new file:   hw2/spades/K33/configs/config.info\n",
    "        new file:   hw2/spades/K33/configs/construction.info\n",
    "        new file:   hw2/spades/K33/configs/detail_info_printer.info\n",
    "        new file:   hw2/spades/K33/configs/distance_estimation.info\n",
    "        new file:   hw2/spades/K33/configs/hmm_mode.info\n",
    "        new file:   hw2/spades/K33/configs/isolate_mode.info\n",
    "        new file:   hw2/spades/K33/configs/large_genome_mode.info\n",
    "        new file:   hw2/spades/K33/configs/mda_mode.info\n",
    "        new file:   hw2/spades/K33/configs/meta_mode.info\n",
    "        new file:   hw2/spades/K33/configs/metaplasmid_mode.info\n",
    "        new file:   hw2/spades/K33/configs/metaviral_mode.info\n",
    "        new file:   hw2/spades/K33/configs/pe_params.info\n",
    "        new file:   hw2/spades/K33/configs/plasmid_mode.info\n",
    "        new file:   hw2/spades/K33/configs/rna_mode.info\n",
    "        new file:   hw2/spades/K33/configs/rnaviral_mode.info\n",
    "        new file:   hw2/spades/K33/configs/sewage_mode.info\n",
    "        new file:   hw2/spades/K33/configs/simplification.info\n",
    "        new file:   hw2/spades/K33/configs/toy.info\n",
    "        new file:   hw2/spades/K33/final.lib_data\n",
    "        new file:   hw2/spades/K33/simplified_contigs/contigs.off\n",
    "        new file:   hw2/spades/K33/simplified_contigs/contigs.seq\n",
    "        new file:   hw2/spades/K33/simplified_contigs/contigs_info\n",
    "        new file:   hw2/spades/K55/assembly_graph.fastg\n",
    "        new file:   hw2/spades/K55/assembly_graph_after_simplification.gfa\n",
    "        new file:   hw2/spades/K55/assembly_graph_with_scaffolds.gfa\n",
    "        new file:   hw2/spades/K55/before_rr.fasta\n",
    "        new file:   hw2/spades/K55/configs/careful_mda_mode.info\n",
    "        new file:   hw2/spades/K55/configs/careful_mode.info\n",
    "        new file:   hw2/spades/K55/configs/config.info\n",
    "        new file:   hw2/spades/K55/configs/construction.info\n",
    "        new file:   hw2/spades/K55/configs/detail_info_printer.info\n",
    "        new file:   hw2/spades/K55/configs/distance_estimation.info\n",
    "        new file:   hw2/spades/K55/configs/hmm_mode.info\n",
    "        new file:   hw2/spades/K55/configs/isolate_mode.info\n",
    "        new file:   hw2/spades/K55/configs/large_genome_mode.info\n",
    "        new file:   hw2/spades/K55/configs/mda_mode.info\n",
    "        new file:   hw2/spades/K55/configs/meta_mode.info\n",
    "        new file:   hw2/spades/K55/configs/metaplasmid_mode.info\n",
    "        new file:   hw2/spades/K55/configs/metaviral_mode.info\n",
    "        new file:   hw2/spades/K55/configs/pe_params.info\n",
    "        new file:   hw2/spades/K55/configs/plasmid_mode.info\n",
    "        new file:   hw2/spades/K55/configs/rna_mode.info\n",
    "        new file:   hw2/spades/K55/configs/rnaviral_mode.info\n",
    "        new file:   hw2/spades/K55/configs/sewage_mode.info\n",
    "        new file:   hw2/spades/K55/configs/simplification.info\n",
    "        new file:   hw2/spades/K55/configs/toy.info\n",
    "        new file:   hw2/spades/K55/final.lib_data\n",
    "        new file:   hw2/spades/K55/final_contigs.fasta\n",
    "        new file:   hw2/spades/K55/final_contigs.paths\n",
    "        new file:   hw2/spades/K55/scaffolds.fasta\n",
    "        new file:   hw2/spades/K55/scaffolds.paths\n",
    "        new file:   hw2/spades/assembly_graph.fastg\n",
    "        new file:   hw2/spades/assembly_graph_after_simplification.gfa\n",
    "        new file:   hw2/spades/assembly_graph_with_scaffolds.gfa\n",
    "        new file:   hw2/spades/before_rr.fasta\n",
    "        new file:   hw2/spades/contigs.fasta\n",
    "        new file:   hw2/spades/contigs.paths\n",
    "        new file:   hw2/spades/corrected/SRR292678sub_S1_L001_R1_00100.0_0.cor.fastq.gz\n",
    "        new file:   hw2/spades/corrected/SRR292678sub_S1_L001_R2_00100.0_0.cor.fastq.gz\n",
    "        new file:   hw2/spades/corrected/SRR292678sub_S1_L001_R_unpaired00.0_0.cor.fastq.gz\n",
    "        new file:   hw2/spades/corrected/configs/config.info\n",
    "        new file:   hw2/spades/corrected/corrected.yaml\n",
    "        new file:   hw2/spades/dataset.info\n",
    "        new file:   hw2/spades/input_dataset.yaml\n",
    "        new file:   hw2/spades/misc/broken_scaffolds.fasta\n",
    "        new file:   hw2/spades/params.txt\n",
    "        new file:   hw2/spades/pipeline_state/stage_0_before_start\n",
    "        new file:   hw2/spades/pipeline_state/stage_10_as_finish\n",
    "        new file:   hw2/spades/pipeline_state/stage_11_bs\n",
    "        new file:   hw2/spades/pipeline_state/stage_12_terminate\n",
    "        new file:   hw2/spades/pipeline_state/stage_1_ec_start\n",
    "        new file:   hw2/spades/pipeline_state/stage_2_ec_runtool\n",
    "        new file:   hw2/spades/pipeline_state/stage_3_ec_compress\n",
    "        new file:   hw2/spades/pipeline_state/stage_4_ec_finish\n",
    "        new file:   hw2/spades/pipeline_state/stage_5_as_start\n",
    "        new file:   hw2/spades/pipeline_state/stage_6_k21\n",
    "        new file:   hw2/spades/pipeline_state/stage_7_k33\n",
    "        new file:   hw2/spades/pipeline_state/stage_8_k55\n",
    "        new file:   hw2/spades/pipeline_state/stage_9_copy_files\n",
    "        new file:   hw2/spades/run_spades.sh\n",
    "        new file:   hw2/spades/run_spades.yaml\n",
    "        new file:   hw2/spades/scaffolds.fasta\n",
    "        new file:   hw2/spades/scaffolds.paths\n",
    "\n",
    "Changes not staged for commit:\n",
    "  (use \"git add <file>...\" to update what will be committed)\n",
    "  (use \"git restore <file>...\" to discard changes in working directory)\n",
    "        modified:   hw2/lab_journal.ipynb\n",
    "        modified:   hw2/spades/corrected/SRR292678sub_S1_L001_R_unpaired00.0_0.cor.fastq.gz\n",
    "\n",
    "(DD_spades_h) mhprs@Seraph:~/PISH/Data_driven/DD_2024_private$\n",
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e8c04f36-acc7-416d-a455-61f6e2821413",
   "metadata": {},
   "source": [
    "### Шаг 3: Оценка качества сборки (15 баллов)\n",
    "Создайте conda-окружение quast\n",
    "\n",
    "Установите QUAST\n",
    "\n",
    "    Проанализируйте оба файла одной командой: contigs.fasta и scaffolds.fasta\n",
    "    Сохраните отчеты в папку hw2/quast\n",
    "    Зафиксируйте команды и основные метрики в lab_journal.ipynb\n",
    "    Сделайте коммит всех изменений\n",
    "\n",
    "Проверка:\n",
    "\n",
    "    - Наличие отчетов QUAST в hw2/quast\n",
    "    - Корректные метрики есть в тексте lab_journal.ipynb (N50, общая длина, количество контигов) и соответствуют результатам запуска QUAST\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4589abd0-ec59-49c7-989f-47e45d93bcb8",
   "metadata": {},
   "source": [
    "Создаём окружение\n",
    "\n",
    "\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
